odl/discr/discr_utils.py

```diff
@@ -817,6 +838,555 @@ def __init__(self, coord_vecs, values, input_type):
         )
 
 
+def _check_func_out_arg(func):
+    """Check if ``func`` has an (optional) ``out`` argument.
```

Comment:
Mention `func_out_type` and that this function only works for functions and `callable`s